### PR TITLE
script to clone code

### PR DESCRIPTION
**Pull request by tst-basilhuffman**
re-write of a similar script of pavel's in the docker build, but a bit
more user friendly. low priority. addresses #540. might be useful
for new folks

### CODE REVIEW

**Comment by tst-basilhuffman** [comment]
refactored to allow for interactive and non interactive mode

to test:

1) ./pull-gee-repository.sh
        clones main repo into "earthenterprise"

2) ./pull-gee-repository.sh -i
        runs script in user-interactive mode

3) ./pull-gee-repository.sh -d <DIR>
        clones main repo into "<DIR>"

4) ./pull-gee-repository.sh -f <USERNAME>
        clones fork <USERNAME> via ssh. can combine with -d

5) ./pull-gee-repository.sh -f <USERNAME> -p <PASSWORD>
        clones fork <USERNAME> via https with password <PASSWORD>

6) ./pull-gee-repository.sh -p <PASSWORD>
        FAILS due to not have a username i.e. fork

7) ./pull-gee-repository.sh -d
        FAILS due to not specifying a directory

8) ./pull-gee-repository.sh -h
        displays help info

File: scripts/pull-gee-repository.sh [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+
+#===  FUNCTION  ================================================================
+#          NAME:  get_type 
+#   DESCRIPTION:  returns the repo to clone, main or fork 
+#    PARAMETERS:  n/a
+#       RETURNS:  IS_FORK, if fork; IS_MAIN, if main
+#===============================================================================
+function get_type ()
+{
+    printf "Do you want to clone the (m)ain repo or a (f)ork (default is main)? "
```
Comment (tst-jblair) [inline scripts/pull-gee-repository.sh]: This reads to me like "main" would be a valid input here... but the check a couple of lines down will *only* match if someone *exactly* enters an "m".  I'm not terribly proficient with bash, is there a smoother way to do this?

File: scripts/pull-gee-repository.sh [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+THIS=$(basename "$0")
+USER_NAME=""
+PASSWORD=""
+DIR_NAME="earthenterprise"
+IS_SSH=1
+IS_HTTPS=2
+IS_MAIN=3
+IS_FORK=4
+SSH_DIR=".ssh"
+SSH_KEY_PRIV="id_rsa"
```
Comment (tst-jblair) [inline scripts/pull-gee-repository.sh]: While I'd believe that RSA is the dominant key type choice, it's not the only choice... I'd expect ECDSA to be a non-zero proportion of the user base?

File: scripts/pull-gee-repository.sh [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+    CLONE_REPO=$?
+
+    if is_cloned
+    then
+        echo "$DIR_NAME is already cloned!"
+        exit 1 
+    fi
+    
+    # if cloning main, just use standard https
+    if [ $CLONE_REPO -eq $IS_MAIN ]
```
Comment (tst-jblair) [inline scripts/pull-gee-repository.sh]: I'm curious why "cloning main" ==> "want to use https"?  Personally, I only ever want to use ssh, to save on the number of passwords I need to keep up with.

Also, why does the path a little lower go through get_password, but this one doesn't?

Comment (tst-basilhuffman) [reply scripts/pull-gee-repository.sh]: This was talked about it slack, current build instructions list as ssh but it was figured that https would be best if you are only cloning the main (google/earthenterprise) repo.  Just covering the case where the user does not have a github account and just wants to pull down the repo.  Thus, no password is required.

Cloning a fork is a different case

File: scripts/pull-gee-repository.sh (hunk 2 above)
Comment (tst-basilhuffman) [reply scripts/pull-gee-repository.sh]: I suppose that the most appropriate validation would be to check that ~/.ssh is present and contains files?

File: scripts/pull-gee-repository.sh (hunk 3 above)
Comment (tst-jblair) [reply scripts/pull-gee-repository.sh]: Ah, interesting - I hadn't recalled that "anonymous https" was an option ;).

File: scripts/pull-gee-repository.sh (hunk 2 above)
Comment (tst-ppenev) [reply scripts/pull-gee-repository.sh]: I don't see the benefit of checking what SSH keys a user has.  I think it's sufficient to use an SSH Git URL, and let `git` handle the rest.

Comment (tst-jblair) [reply scripts/pull-gee-repository.sh]: The specific keys aren't being checked at all; this section is really aimed at seeing *if* the user has keys. My comment really came down to that "id_rsa" is too specific.

Comment (tst-basilhuffman) [reply scripts/pull-gee-repository.sh]: Thats fine, I just figured the standard git error message here is a bit cryptic, doesn't really explain much at first glance

Comment (tst-ppenev) [reply scripts/pull-gee-repository.sh]: I know. And I don't see how checking if a user has keys gives us anything. It seems to be used only to replace the error message that `git` would give you, if you tried to clone over SSH, and Git couldn't find your keys.

Comment (tst-basilhuffman) [reply scripts/pull-gee-repository.sh]: Fair enough. If you think thats sufficient than I can take it out

**Pull request merged**